solr/solrj/src/java/org/apache/solr/client/solrj/request/QueryRequest.java

```diff
@@ -76,5 +76,9 @@ public SolrParams getParams() {
     return query;
   }
 
+  @Override
+  public String getRequestType() {
+    return SolrRequestType.QUERY.toString();
```

Comment:
Something about this that I think bothers me is that we don't make a differentiation between distinct handlers, so we can't use this to build QoS levels. Like I can't say that RTG should always go through while `/select` might be acceptable to throttle and `/spellcheck` is even lower priority.

Reply:
This seems like a query prioritization model IMO -- a good idea. I will open a follow up JIRA once this goes in.